mls-rs/src/group/message_processor.rs

```diff
@@ -668,17 +670,6 @@ pub(crate) trait MessageProcessor: Send + Sync {
             });
         }
 
-        let commit_effect =
```

Comment:
Is the switch in order here so that if the update path doesn't validate we don't waste our time calculating the other stuff?

Reply:
That's a good reason but also we borrow `self` as immutable in `let group_state = self.group_state()` in line 626 so we can't call `self.group_state_mut()` to pop the reinit here.